File: ws/src/main/java/workspace/ws/ds/data/Stack.java
package workspace.ws.ds.data;

import java.util.ArrayList;
import java.util.List;

public class Stack {
	private List<Object> elements;

	public Stack() {
		elements = new ArrayList<Object>();
	}
	
	public void printStack() {
		for (Object element : elements) {
			System.out.println(element);
		}
	}

	public void push(Object object) {
		elements.add(object);
	}

	public Object pop() {
		if (elements.size() < 1) {
			throw new NullPointerException("No more elements");
		}

		return elements.remove(elements.size() - 1);
	}
	
	public boolean isEmpty() {
		return elements.size() == 0;
	}
	
	public Object peek() {
		if (elements.size() < 1) {
			throw new NullPointerException("No more elements");
		}
		
		return elements.get(elements.size() - 1);
	}
}

File: WS/G4G/Problems/stack/stack.py
class Stack(object):
    def __init__(self):
        self.items = []

    def is_empty(self):
        return self.items == []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        return self.items.pop()

    def peek(self):
        if self.is_empty():
            return None

        return self.items[len(self.items) - 1]

    def size(self):
        return len(self.items)

    def get_list(self):
        return self.items
